Src/FluentAssertions.Net40/Xml/XmlElementAssertions.cs

```diff
@@ -0,0 +1,190 @@
+        /// with the specified <paramref name="expectedName"/>, <param name="expectedNamespace"/>
+        /// and <paramref name="expectedValue"/>.
+        /// </summary>
+        /// <param name="expectedName">The name of the expected attribute</param>
+        /// <param name="expectedValue">The value of the expected attribute</param>
+        public AndConstraint<XmlElementAssertions> HaveAttributeWithNamespace(string expectedName, string expectedNamespace, string expectedValue)
+        {
+            return HaveAttributeWithNamespace(expectedName, expectedNamespace, expectedValue, string.Empty);
+        }
+
+        /// <summary>
+        /// Asserts that the current <see cref="XmlElement"/> has an attribute
+        /// with the specified <paramref name="expectedName"/>, <param name="expectedNamespace"/>
+        /// and <paramref name="expectedValue"/>.
+        /// </summary>
+        /// <param name="expectedName">The name of the expected attribute</param>
+        /// <param name="expectedValue">The value of the expected attribute</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion 
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndConstraint<XmlElementAssertions> HaveAttributeWithNamespace(
+            string expectedName,
+            string expectedNamespace,
+            string expectedValue,
+            string because, params object[] becauseArgs)
+        {
+            XmlAttribute attribute = Subject.Attributes[expectedName, expectedNamespace];
+
+            string expectedFormattedName =
+                (string.IsNullOrEmpty(expectedNamespace) ? "" : "{" + expectedNamespace + "}")
+                + expectedName;
+
+            Execute.Assertion
+                .ForCondition(attribute != null)
+                .BecauseOf(because, becauseArgs)
+                .FailWith(
+                    "Expected XML element to have attribute {0}" 
+                    + " with value {1}{reason}, but found no such attribute in {2}",
+                    expectedFormattedName, expectedValue, Subject);
+
+            Execute.Assertion
+                .ForCondition(attribute.Value == expectedValue)
+                .BecauseOf(because, becauseArgs)
+                .FailWith(
+                    "Expected XML attribute {0} to have value {1}{reason}, but found {2}.",
+                    expectedFormattedName, expectedValue, attribute.Value);
+
+            return new AndConstraint<XmlElementAssertions>(this);
+        }
+
+        public AndWhichConstraint<XmlElementAssertions, XmlElement> HaveElement(string expectedElement)
```

Comment:
Sorry, I think I was a bit too much in a hurry when trying to finish this. Adding all the missing docs.